src/main/java/net/snowflake/ingest/streaming/internal/RowBuffer.java

```diff
@@ -0,0 +1,60 @@
+package net.snowflake.ingest.streaming.internal;
+
+import java.util.List;
+import java.util.Map;
+import net.snowflake.ingest.streaming.InsertValidationResponse;
+import net.snowflake.ingest.utils.Constants;
+
+/**
```

Comment:
so far yes, I find it more readable, because interface describes what is really needed by the using classes.
Also states what to test, I usually considered it as a best practice in java, less implicit dependencies and allows to track single responsibilities.